### PR TITLE
#167730729 configure eslint for the project

### PR DESCRIPTION
**Pull request by Dolapo47**
#### What does this PR do?
* configure eslint for the project (167730729)
#### Description of Task to be completed?
* install the necessary dependencies (eslint, eslint-config-airbnb-base, eslint-plugin-import)
#### How should this be manually tested?
* git clone this branch <https://github.com/andela/firestar-backend.git>
* cd to the project `firestar-backend`
* Checkout to the branch `ch-configure-eslint-#167730729`
* run 'npm install'
* run 'npm run lint' on the terminal. it fixes all linting issues and lets you know what it cannot resolve

#### Any background context you want to provide?
* N/A
#### What are the relevant pivotal tracker stories?
* [PT story ID](https://www.pivotaltracker.com/story/show/167730729)
#### Screenshots or gifs (if appropriate)
Screenshot (3)

#### Questions:
* N/A

### CODE REVIEW

**Review by wahabtobibello — APPROVED** [review]

**Review by marusoft — CHANGES_REQUESTED** [review]
Great job @Dolapo47, You need to make a few changes to the where someone can manually test the branch.
1. add a link to the project
2 and the branch

**Comment by Dolapo47** [comment]
> Great job @Dolapo47, You need to make a few changes to the where someone can manually test the branch.
> 
> 1. add a link to the project
>    2 and the branch

worked on those changes, kindly review

**Review by marusoft — APPROVED** [review]
Awesome @Dolapo47

File: src/models/User.js [hunk 1]
```diff
@@ -1,57 +1,57 @@
...
+    .toString('hex');
+  return this.hash === hash;
 };
 
-UserSchema.methods.setPassword = function(password) {
-    this.salt = crypto.randomBytes(16).toString("hex");
-    this.hash = crypto
-        .pbkdf2Sync(password, this.salt, 10000, 512, "sha512")
-        .toString("hex");
+UserSchema.methods.setPassword = function (password) {
```
[the remark this answers is not in the document]
Comment (rovilay) [reply src/models/User.js]: @Dolapo47  Please use es6 function `() => {}` .

File: src/index.js [hunk 2]
```diff
@@ -1,93 +1,95 @@
...
-    cors = require("cors"),
-    passport = require("passport"),
-    errorhandler = require("errorhandler"),
-    mongoose = require("mongoose");
-
-const isProduction = process.env.NODE_ENV === "production";
+const fs = require('fs'),
+  http = require('http'),
+  path = require('path'),
+  methods = require('methods'),
```
[the remark this answers is not in the document]
Comment (rovilay) [reply src/index.js]: Please resolve.

File: src/index.js [hunk 3]
```diff
@@ -1,93 +1,95 @@
...
-    session = require("express-session"),
-    cors = require("cors"),
-    passport = require("passport"),
-    errorhandler = require("errorhandler"),
-    mongoose = require("mongoose");
-
-const isProduction = process.env.NODE_ENV === "production";
+const fs = require('fs'),
+  http = require('http'),
+  path = require('path'),
```
[the remark this answers is not in the document]
Comment (rovilay) [reply src/index.js]: Please remove all unused variables.

File: src/index.js [hunk 4]
```diff
@@ -1,93 +1,95 @@
...
-    bodyParser = require("body-parser"),
-    session = require("express-session"),
-    cors = require("cors"),
-    passport = require("passport"),
-    errorhandler = require("errorhandler"),
-    mongoose = require("mongoose");
-
-const isProduction = process.env.NODE_ENV === "production";
+const fs = require('fs'),
+  http = require('http'),
```
[the remark this answers is not in the document]
Comment (rovilay) [reply src/index.js]: Please remove all unused variables.

**Comment by rovilay** [comment]
@Dolapo47 I keep getting this error when I run `npm run lint`.
Screenshot 2019-08-16 at 1 33 55 PM

**Comment by rovilay** [comment]
@Dolapo47 Please use the suggested PR title format.

**Comment by Dolapo47** [comment]
> @Dolapo47 I keep getting this error when I run `npm run lint`.
> Screenshot 2019-08-16 at 1 33 55 PM

Thanks for the feedback
I did a fresh git clone on another system; it ran as expected
dolapo

**Comment by Dolapo47** [comment]
> @Dolapo47 Please use the suggested PR title format.

Thanks for the feedback
PR title format has been treated

**Comment by Dolapo47** [comment]
> > @Dolapo47 I keep getting this error when I run `npm run lint`.
> > Screenshot 2019-08-16 at 1 33 55 PM
> 
> Thanks for the feedback
> I did a fresh git clone on another system; it ran as expected
> dolapo

I have corrected the lint script, kindly review

File: src/models/User.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (Dolapo47) [reply src/models/User.js]: Treated: converted common js function to es6

File: src/routes/api/index.js [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
-router.use(function(err, req, res, next) {
-    if (err.name === "ValidationError") {
-        return res.status(422).json({
-            errors: Object.keys(err.errors).reduce(function(errors, key) {
-                errors[key] = err.errors[key].message;
-                return errors;
-            }, {})
-        });
-    }
+router.use((err, req, res, next) => {
```
[the remark this answers is not in the document]
Comment (rovilay) [reply src/routes/api/index.js:8]: @Dolapo47 Good work. Can you reachout to adedeko to fix the hound issues.

**Comment by rovilay** [comment]
@Dolapo47 Please resolve conflict

**Comment by rovilay** [comment]
@Dolapo47 Please resolve conflicts

File: package.json [hunk 6]
```diff
@@ -36,6 +36,9 @@
     "underscore": "^1.9.1"
   },
   "devDependencies": {
+    "eslint": "^6.1.0",
```
Comment (wahabtobibello) [inline package.json:41]: The `lint` command has been removed

**Review by wahabtobibello — CHANGES_REQUESTED** [review]

**Comment by wahabtobibello** [comment]
@Dolapo47 I want to work on fixing the eslint errors. Is that cool?

File: package.json (hunk 6 above)
Comment (Dolapo47) [reply package.json:41]: ok

File: src/models/User.js [hunk 7]
```diff
@@ -4,54 +4,52 @@ import crypto from 'crypto';
...
+    hash: String,
+    salt: String
+  },
+  { timestamps: true }
 );
 
-UserSchema.plugin(uniqueValidator, { message: "is already taken." });
+UserSchema.plugin(uniqueValidator, { message: 'is already taken.' });
 
-UserSchema.methods.validPassword = function(password) {
```
Comment (wahabtobibello) [inline src/models/User.js:36]: This change here is going to lead to a bug. Arrow functions do not have the `this` reference so `this.hash` cannot be read. Learn [more](https://medium.com/@lucasdavidferrero/dont-use-arrow-functions-when-you-use-mongoose-schema-method-190b79f1640c).

File: src/models/User.js [hunk 8]
```diff
@@ -4,54 +4,52 @@ import crypto from 'crypto';
...
-        .toString("hex");
-    return this.hash === hash;
+UserSchema.methods.validPassword = (password) => {
+  const hash = crypto
+    .pbkdf2Sync(password, this.salt, 10000, 512, 'sha512')
+    .toString('hex');
+  return this.hash === hash;
 };
 
-UserSchema.methods.setPassword = function(password) {
```
Comment (wahabtobibello) [inline src/models/User.js:43]: This change here is going to lead to a bug. Arrow functions do not have the `this` reference so `this.hash` and `this.salt` cannot be assigned. Learn [more](https://medium.com/@lucasdavidferrero/dont-use-arrow-functions-when-you-use-mongoose-schema-method-190b79f1640c).

File: src/models/User.js [hunk 9]
```diff
@@ -4,54 +4,52 @@ import crypto from 'crypto';
...
-        .pbkdf2Sync(password, this.salt, 10000, 512, "sha512")
-        .toString("hex");
+UserSchema.methods.setPassword = (password) => {
+  this.salt = crypto.randomBytes(16).toString('hex');
+  this.hash = crypto
+    .pbkdf2Sync(password, this.salt, 10000, 512, 'sha512')
+    .toString('hex');
 };
 
-UserSchema.methods.toAuthJSON = function() {
```
Comment (wahabtobibello) [inline src/models/User.js:50]: This change here is going to lead to a bug. Arrow functions do not have the `this` reference so `this.username` and `this.email` cannot be assigned. Learn [more](https://medium.com/@lucasdavidferrero/dont-use-arrow-functions-when-you-use-mongoose-schema-method-190b79f1640c).

**Review by wahabtobibello — CHANGES_REQUESTED** [review]

File: src/models/User.js (hunk 9 above)
Comment (wahabtobibello) [reply src/models/User.js:50]: Kindly change this back to the `function(){}` declaration

File: src/models/User.js (hunk 7 above)
Comment (wahabtobibello) [reply src/models/User.js:36]: Kindly change this back to the `function(){}` declaration

File: src/models/User.js (hunk 8 above)
Comment (wahabtobibello) [reply src/models/User.js:43]: Kindly change this back to the `function(){}` declaration

Comment (Dolapo47) [reply src/models/User.js:43]: some team mates have been assigned the task of creating db and jwt, whoever is assigned that ticket will handle that

File: src/models/User.js (hunk 7 above)
Comment (Dolapo47) [reply src/models/User.js:36]: some team mates have been assigned the task of creating db and jwt, whoever is assigned that ticket will handle that

File: src/models/User.js (hunk 9 above)
Comment (Dolapo47) [reply src/models/User.js:50]: some team mates have been assigned the task of creating db and jwt, whoever is assigned that ticket will handle that

File: src/models/User.js (hunk 8 above)
Comment (wahaaabello) [reply src/models/User.js]: OK then

**Review by wahabtobibello — APPROVED** [review]

**Review by rovilay — APPROVED** [review]

**Pull request merged**